test/e2e/e2e_test.go

```diff
@@ -415,6 +416,37 @@ var _ = Describe("controller", Ordered, func() {
 			}
 			EventuallyWithOffset(1, verifyPdbNotExists, time.Minute, time.Second).Should(Succeed())
 			EventuallyWithOffset(1, verifyEvictionAutoScalerNotExists, time.Minute, time.Second).Should(Succeed())
+
+			By("Scraping controller metrics at the end of the e2e test")
+
+			scrapeMetrics := func() error {
+				// Fetch controller-manager pod name via kubectl and label selector
+				cmd := exec.Command("kubectl", "-n", "kube-system", "get", "pods", "-l", "app.kubernetes.io/name=eviction-autoscaler", "-o", "jsonpath={.items[0].metadata.name}")
```

Comment:
not mandatory  but you can also use clientset to proxy instead of shelling out to kubectl. Shelling out can be a bit harder to error handle sometimes.

Reply:
A fun chacne later to learn about go routines and cancelation.